### PR TITLE
Add Native LagCompensation plugin for Rocks

### PR DESCRIPTION
**Pull request by ProdigySim**
Adds a plugin to enable native LagCompensation on tank rocks.

I got this tip from @domino on AlliedModders discord. We can use `CLagCompensationManager::AddAdditionalEntity()` to add tank rocks to the lag compesation tracking. This seems to work with `net_fakelag`, but it does visually play the rock detonation animation at a different location. That is, rocks appear to get teleported back to where the player shot them to explode.

This has a few advantages over the existing rock lag compensation plugin:
- Native damage calculations
- Native ray tracing & penetration mechanics
- Native spread computations
- Native rock hitboxes

Of course, this might be a significant balance change which would make rocks harder to skeet again if included in Zonemod or other configs.

For now, I will just open this PR to document the plugin.

### CODE REVIEW

File: addons/sourcemod/scripting/l4d2_lagcomp_manager.sp [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	StartPrepSDKCall(SDKCall_Raw);
+	PrepSDKCall_SetFromConf(hGameConf, SDKConf_Signature, "CLagCompensationManager_AddAdditionaEntity");
+	PrepSDKCall_AddParameter(SDKType_CBaseEntity, SDKPass_Pointer);
+	g_hLagCompAddEntity = EndPrepSDKCall();
+
+	StartPrepSDKCall(SDKCall_Raw);
+	PrepSDKCall_SetFromConf(hGameConf, SDKConf_Signature, "CLagCompensationManager_RemoveAdditionaEntity");
+	PrepSDKCall_AddParameter(SDKType_CBaseEntity, SDKPass_Pointer);
+	g_hLagCompRemoveEntity = EndPrepSDKCall();
+	// PrintToServer("Values: 0x%08x, %08x, %08x", g_lagcompensation, g_hLagCompAddEntity, g_hLagCompRemoveEntity);
```
Comment (ProdigySim) [inline addons/sourcemod/scripting/l4d2_lagcomp_manager.sp:31]: TODO: Add a SetFailState or similar so we don't have to go searching if these signatures fail?

File: addons/sourcemod/scripting/l4d2_lagcomp_manager.sp [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+{
+	if (StrEqual(classname, "tank_rock")) {
+		SDKCall(g_hLagCompAddEntity, g_lagcompensation, entity);
+	}
+}
+
+public void OnEntityDestroyed(int entity)
+{
+	if (IsRock(entity)) {
+		SDKCall(g_hLagCompRemoveEntity, g_lagcompensation, entity);
```
Comment (ProdigySim) [inline addons/sourcemod/scripting/l4d2_lagcomp_manager.sp:44]: Possible edge case? What if the plugin is unloaded, or map is changed, while rock is still alive?

**Comment by ProdigySim** [comment]
Did a 1 map test pug and some messing around with it in cheats mode with some pro players. The players on 110-180 ping said it was okay after some training on it. Adam said it was bad. Personally I feel like the hitbox accuracy is the biggest nerf, since rocks leave the tank's hands with random rotation. Side skeets with smg can be very hard for some rock orientations.

File: addons/sourcemod/scripting/l4d2_lagcomp_manager.sp (hunk 2 above)
Comment (ProdigySim) [reply addons/sourcemod/scripting/l4d2_lagcomp_manager.sp:46]: I read more of the code for LagCompensation, and it looks like the whole Lagcomp system gets destroyed & re-created on amp change, so we shouldn't really have to worry about leaks here. Also, if an entity doesn't exist, this "remove" call just silently does nothing.
